lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/DBODoiDaoImpl.java

```diff
@@ -104,60 +89,68 @@ public Doi createDoi(final String userGroupId, final String objectId,
 		}
-		if (objectType == null) {
+		if (dto.getObjectType() == null) {
 			throw new IllegalArgumentException("Object type cannot be null.");
 		}
-		if (doiStatus == null) {
+		if (dto.getDoiStatus() == null) {
 			throw new IllegalArgumentException("DOI status cannot be null.");
 		}
+		DBODoi dbo = getDbo(dto);
 
-		DBODoi dbo = getDbo(objectId, objectType, versionNumber);
-		if (!dbo.getETag().equals(etag)) {
+		if (!dbo.getETag().equals(dto.getEtag())) {
 			throw new ConflictingUpdateException("Etags do not match.");
 		}
-		dbo.setDoiStatus(doiStatus);
+		dbo.setDoiStatus(dto.getDoiStatus());
+		dbo.setETag(UUID.randomUUID().toString());
 		boolean success = basicDao.update(dbo);
 		if (!success) {
 			throw new DatastoreException("Update failed for " + dbo);
 		}
-		return getDoi(objectId, objectType, versionNumber);
+		return getDoi(dto);
 	}
 
 	@Override
-	public Doi getDoi(final String objectId, final ObjectType objectType,
-			final Long versionNumber) throws NotFoundException, DatastoreException {
+	public Doi getDoi(Doi dto) throws NotFoundException, DatastoreException {
 
-		if (objectId == null || objectId.isEmpty()) {
+		if (dto.getObjectId() == null) {
 			throw new IllegalArgumentException("Object ID cannot be null nor empty.");
 		}
-		if (objectType == null) {
+		if (dto.getObjectType() == null) {
 			throw new IllegalArgumentException("Object type cannot be null.");
 		}
-		DBODoi dbo = getDbo(objectId, objectType, versionNumber);
+		DBODoi dbo = null;
+		dbo = getDbo(dto);
 		return DoiUtils.convertToDto(dbo);
 	}
 
-	private DBODoi getDbo (String objectId, ObjectType objectType, Long versionNumber)
+	/**
+	 * Retrieves a DOI DBO corresponding to the DTO supplied as a parameter.
+ 	 * @param dto The DTO containing object ID, type, and version number that
+	 *            corresponds to an existing DBO.
+	 * @return The existing DBO.
+	 * @throws NotFoundException The DOI was not found in the database
+	 * @throws DatastoreException More than one DOI was found when exactly
+	 *                            one should have been found.
+	 */
+	private DBODoi getDbo (Doi dto)
 			throws NotFoundException, DatastoreException {
+		DBODoi translatedDbo = convertToDbo(dto);
 
-		String sql = (versionNumber == null ? SELECT_DOI_NULL_OBJECT_VERSION : SELECT_DOI);
+		String sql = SELECT_DOI;
 		MapSqlParameterSource paramMap = new MapSqlParameterSource();
-		paramMap.addValue(COL_DOI_OBJECT_ID, KeyFactory.stringToKey(objectId));
-		paramMap.addValue(COL_DOI_OBJECT_TYPE, objectType.name());
-		if (versionNumber != null) {
-			paramMap.addValue(COL_DOI_OBJECT_VERSION, versionNumber);
-		}
+		paramMap.addValue(COL_DOI_OBJECT_ID, translatedDbo.getObjectId());
+		paramMap.addValue(COL_DOI_OBJECT_TYPE, translatedDbo.getObjectType());
+		paramMap.addValue(COL_DOI_OBJECT_VERSION, translatedDbo.getObjectVersion());
 		List<DBODoi> dboList = namedJdbcTemplate.query(sql, paramMap, rowMapper);
```

Comment:
namedJdbcTemplate.queryObject(sql, paramMap, rowMapper) 
catch Springs exception when it encounters anything other than one, and rethorw IllegalState()